pkg/controller/import-controller.go

```diff
@@ -72,6 +93,56 @@ func (ic *ImportController) findImportPodFromCache(pvc *v1.PersistentVolumeClaim
 	return podList[0], nil
 }
 
+func (ic *ImportController) importerRequiresScratchSpace(pvc *v1.PersistentVolumeClaim) bool {
+	contentType := getContentType(pvc)
+	// All archive requires scratch space.
+	if contentType == "archive" {
+		glog.V(3).Infof("Archive content type, requires scratch space\n")
+		return true
+	}
+	source := getSource(pvc)
+	// Invalid Sources will have gotten rejected by the web hook, so this is fine.
+	if source != SourceHTTP && source != SourceNone {
+		glog.V(3).Infof("Source type %s, requires scratch space\n", source)
+		return true
+	}
+	return false
+}
+
+// getScratchPvcStorageClass tries to determine which storage class to use for use with a scratch persistent
+// volume claim. The order of preference is the following:
+// 1. Defined value in CDI config map.
+// 2. If 1 is not available use the 'default' storage class.
+// 3. If 2 is not available use the storage class name of the pvc that will own the scratch pvc.
+// 4. If none of those are available, fail with an error.
+func (ic *ImportController) getScratchPvcStorageClass(pvc *v1.PersistentVolumeClaim) (string, error) {
+	var storageClassName = ""
+	// TODO: Read config map to see if it has a storageClassName field in it.
+
+	if storageClassName == "" {
```

Comment:
I am not a fan of functions with multiple exit points. So this function is constructed that as soon as storageClassName has a value, all the subsequent if statements will fail and it will get to the exit point at the bottom of the function.

Reply:
Good point, and I am doing something similar above, I should be consistent.